src/main/java/org/wise/portal/presentation/web/controllers/tag/UserTagController.java

```diff
@@ -0,0 +1,85 @@
+package org.wise.portal.presentation.web.controllers.tag;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.wise.portal.domain.user.User;
+import org.wise.portal.domain.usertag.UserTag;
+import org.wise.portal.presentation.web.response.ResponseEntityGenerator;
+import org.wise.portal.service.user.UserService;
+import org.wise.portal.service.usertags.UserTagsService;
+
+@RestController
+@Secured({ "ROLE_TEACHER" })
+@RequestMapping("/api")
+public class UserTagController {
+
+  @Autowired
+  private UserService userService;
+
+  @Autowired
+  private UserTagsService userTagsService;
+
+  @PostMapping("/user/tag")
+  protected ResponseEntity<Map<String, Object>> createTag(Authentication auth,
```

Comment:
Instead of returning ResponseEntity, can we just return the Map object and throw a TagAlreadyExistsException if a tag exists?